integration/test/upgradetiller/upgrade_tiller_test.go

```diff
@@ -86,3 +116,18 @@ func getTillerImage(ctx context.Context, namespace, labelSelector string) (strin
 
 	return tillerImage, nil
 }
+
+func updateTillerImage(ctx context.Context, namespace, labelSelector, tillerImage string) error {
+	d, err := getTillerDeployment(ctx, namespace, labelSelector)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	d.Spec.Template.Spec.Containers[0].Image = tillerImage
+	_, err = config.K8sClient.Apps().Deployments(namespace).Update(d)
```

Comment:
This feels like a hack. I think we can improve the test a lot with very little effort if we make the `tillerImage` a property of the `Client` struct. This should default to the current value so the change isn't breaking. Then you can install "outdated" tiller with `config.HelmClient.EnsureTillerInstalled` set `config.HelmClient.tillerImage` to the newer version and call `config.HelmClient.EnsureTillerInstalled` again. Such test would better reflect the reality because we don't know what update does under the hood.

Reply:
Yes that is cleaner and simplifies the test. Nice that it means the image is set via EnsureTillerInstalled. I'll update.